File: app/src/test/java/com/github/goldy1992/mp3player/client/MockMediaControllerAdapter.java
package com.github.goldy1992.mp3player.client;

import android.content.Context;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.media.MediaMetadataCompat;
import android.support.v4.media.session.MediaSessionCompat;
import android.support.v4.media.session.PlaybackStateCompat;

import com.github.goldy1992.mp3player.client.callbacks.MyMediaControllerCallback;
import com.github.goldy1992.mp3player.client.callbacks.metadata.MetadataListener;
import com.github.goldy1992.mp3player.client.callbacks.playback.PlaybackStateListener;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import static android.support.v4.media.MediaMetadataCompat.METADATA_KEY_ARTIST;
import static android.support.v4.media.MediaMetadataCompat.METADATA_KEY_TITLE;

public class MockMediaControllerAdapter extends MediaControllerAdapter {


    @Inject
    public MockMediaControllerAdapter(Context context, MyMediaControllerCallback myMediaControllerCallback) {
        super(context, myMediaControllerCallback);
    }

    @Override
    public void setMediaToken(MediaSessionCompat.Token token) {
        // DO NOTHING
    }

    @Override
    public void prepareFromMediaId(String mediaId, Bundle extras) {
        // DO NOTHING
    }

    @Override
    public void playFromMediaId(String mediaId, Bundle extras) {
        // DO NOTHING
    }

    @Override
    public void playFromUri(Uri uri, Bundle extras) {
        // DO NOTHING
    }

    @Override
    public void play() {
        // DO NOTHING
    }
    @Override
    public void setRepeatMode(@PlaybackStateCompat.RepeatMode int repeatMode) {
        // DO NOTHING
    }
    @Override
    public void pause() {
        // DO NOTHING
    }
    @Override
    public void seekTo(long position) {
        // DO NOTHING
    }
    @Override
    public void stop() {
        // DO NOTHING
    }
    @Override
    public void skipToNext() {
        // DO NOTHING
    }
    @Override
    public void skipToPrevious() {
        // DO NOTHING
    }
    @Override
    public void setShuffleMode(@PlaybackStateCompat.ShuffleMode int shuffleMode) {
        // DO NOTHING
    }
    @Override
    public void registerMetaDataListener(MetadataListener metaDataListener) {
        // DO NOTHING
    }
    @Override
    public void unregisterMetaDataListener(MetadataListener metaDataListener) {
        // DO NOTHING
    }
    @Override
    public void registerPlaybackStateListener(PlaybackStateListener playbackStateListener) {
        // DO NOTHING
    }

    @Override
    public int getPlaybackState() {
        return PlaybackStateCompat.STATE_NONE;
    }

    @Override
    public PlaybackStateCompat getPlaybackStateCompat() {
        return new PlaybackStateCompat.Builder().setState(getPlaybackState(), 0L, 0F).build();
    }
    @Override
    public MediaMetadataCompat getMetadata() {
        return new MediaMetadataCompat.Builder()
            .putString(METADATA_KEY_ARTIST, "artist")
            .putString(METADATA_KEY_TITLE, "title")
            .build();
    }

    @Override
    public List<MediaSessionCompat.QueueItem> getQueue() {
        return new ArrayList<>();
    }

    @Override
    public long getActiveQueueItemId() {
        return 0L;
    }

    public @PlaybackStateCompat.ShuffleMode int getShuffleMode() {
        return PlaybackStateCompat.SHUFFLE_MODE_ALL;
    }

    public @PlaybackStateCompat.RepeatMode int getRepeatMode() {
        return PlaybackStateCompat.REPEAT_MODE_ALL;
    }

    public MediaSessionCompat.Token getToken() {
        return null;
    }

    public void disconnect() {
        // DO NOTHING
    }

    public boolean isInitialized() {
        return true;
    }


    public void sendCustomAction(String customAction, Bundle args) {
        // DO NOTHING
    }
}

File: client/testsupport/src/main/java/com/github/goldy1992/mp3player/client/testsupport/MockMediaControllerAdapter.kt
package com.github.goldy1992.mp3player.client.testsupport

import android.content.Context
import android.net.Uri
import android.os.Bundle
import android.support.v4.media.MediaMetadataCompat
import android.support.v4.media.session.MediaSessionCompat
import android.support.v4.media.session.PlaybackStateCompat
import android.support.v4.media.session.PlaybackStateCompat.ShuffleMode
import com.github.goldy1992.mp3player.client.MediaControllerAdapter
import com.github.goldy1992.mp3player.client.callbacks.MyMediaControllerCallback
import com.github.goldy1992.mp3player.client.callbacks.metadata.MetadataListener
import com.github.goldy1992.mp3player.client.callbacks.playback.PlaybackStateListener
import java.util.*
import javax.inject.Inject

class MockMediaControllerAdapter
    @Inject
    constructor(context: Context?,
                myMediaControllerCallback: MyMediaControllerCallback?)
    : MediaControllerAdapter(context!!, myMediaControllerCallback) {
    override fun setMediaToken(token: MediaSessionCompat.Token?) { // DO NOTHING
    }

    override fun prepareFromMediaId(mediaId: String?, extras: Bundle?) { // DO NOTHING
    }

    override fun playFromMediaId(mediaId: String?, extras: Bundle?) { // DO NOTHING
    }

    override fun playFromUri(uri: Uri?, extras: Bundle?) { // DO NOTHING
    }

    override fun play() { // DO NOTHING
    }

    override fun pause() { // DO NOTHING
    }

    override fun seekTo(position: Long) { // DO NOTHING
    }

    override fun stop() { // DO NOTHING
    }

    override fun skipToNext() { // DO NOTHING
    }

    override fun skipToPrevious() { // DO NOTHING
    }

    override fun registerMetaDataListener(metaDataListener: MetadataListener?) { // DO NOTHING
    }

    override fun unregisterMetaDataListener(metaDataListener: MetadataListener?) { // DO NOTHING
    }

    override fun registerPlaybackStateListener(playbackStateListener: PlaybackStateListener?) { // DO NOTHING
    }

    override val playbackState: Int
        get() = PlaybackStateCompat.STATE_NONE

    override val playbackStateCompat: PlaybackStateCompat
        get() = PlaybackStateCompat.Builder().setState(playbackState, 0L, 0f).build()

    override val metadata: MediaMetadataCompat
        get() = MediaMetadataCompat.Builder()
                .putString(MediaMetadataCompat.METADATA_KEY_ARTIST, "artist")
                .putString(MediaMetadataCompat.METADATA_KEY_TITLE, "title")
                .build()

    override val queue: List<MediaSessionCompat.QueueItem>
        get() = ArrayList()

    override val activeQueueItemId: Long
        get() = 0L

    // DO NOTHING
    @get:ShuffleMode
    override var shuffleMode: Int
        get() = PlaybackStateCompat.SHUFFLE_MODE_ALL
        set(shuffleMode) { // DO NOTHING
        }

    // DO NOTHING
    @get:PlaybackStateCompat.RepeatMode
    override var repeatMode: Int
        get() = PlaybackStateCompat.REPEAT_MODE_ALL
        set(repeatMode) { // DO NOTHING
        }

    override var token: MediaSessionCompat.Token?
        get() = null
        set(token) {
            super.token = token
        }

    override fun disconnect() { // DO NOTHING
    }

    override val isInitialized: Boolean
        get() = true

    override fun sendCustomAction(customAction: String?, args: Bundle?) { // DO NOTHING
    }
}